libraries/AP_Frsky_Telem/AP_Frsky_Telem.cpp

```diff
@@ -1002,8 +998,372 @@ bool AP_Frsky_Telem::get_telem_data(uint8_t &frame, uint16_t &appid, uint32_t &d
     return singleton->_get_telem_data(frame, appid, data);
 }
 
-namespace AP {
-    AP_Frsky_Telem *frsky_telem() {
-        return AP_Frsky_Telem::get_singleton();
+#if HAL_WITH_FRSKY_TELEM_BIDIRECTIONAL
+/*
+  fetch Sport data for an external transport, such as FPort
+ */
+bool AP_Frsky_Telem::_set_telem_data(const uint8_t frame, const uint16_t appid, const uint32_t data)
+{
+    sport_packet_t sp{};
+
+    sp.sensor = 0x0D;
+    sp.frame = frame;
+    sp.appid = appid;
+    sp.data = data;
```

Comment:
Try to create this as a `const` using the same syntax as in https://github.com/ardupilot/ardupilot/blob//libraries/AP_Logger/LogFile.cpp#L169

It's really good in that it catches "narrowing" errors where you might inadvertently try to shove something to big into a field which is too small.

Reply:
that syntax does not work for typedefs only for plain structs, this works though
```
const sport_packet_t sp {
  0x0D,
  frame,
  appid,
  data
}
```